zebra-consensus/src/transaction.rs

```diff
@@ -186,6 +184,9 @@ where
             if tx.is_coinbase() {
                 check::coinbase_tx_no_prevout_joinsplit_spend(&tx)?;
             }
+            if req.is_mempool() && tx.contains_coinbase_input() {
```

Comment:
Minor - Security: Can we do this rejection as the first check?

In general, we try to do simpler checks before more complicated checks, so we:
* run less code on invalid transactions, decreasing our attack surface
* spend less CPU and RAM on invalid transactions

Reply:
Makes sense, changed in 315eaf53